### PR TITLE
fix: implement `Headers#delete` correctly

### LINKED ISSUE #2429: `Headers#delete` returns `true` if the value exists.
## Bug Description

`Headers#delete` returns `true` if the value exists.

## Reproducible By

```js
console.log(new Headers({ "x": "0" }).delete("x")); // true
console.log(new Headers().delete("x")); // undefined
```

## Expected Behavior

```js
console.log(new Headers({ "x": "0" }).delete("x")); // undefined
console.log(new Headers().delete("x")); // undefined
```

## Logs & Screenshots

```shell
> $ node
Welcome to Node.js v21.1.0.
Type ".help" for more information.
> new Headers({ "x": "0" }).delete("x")
true
> new Headers().delete("x")
undefined
>

```

### PR DESCRIPTION
**Pull request by tsctx**
## This relates to...

Fixes #2429 

### Bug Fixes

#2429 

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [x] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/fetch/headers.js [hunk 1]
```diff
@@ -301,14 +301,10 @@ class Headers {
 
     // 6. If this’s header list does not contain name, then
     //    return.
-    if (!this[kHeadersList].contains(name)) {
-      return
-    }
-
```
Comment (ronag) [inline lib/fetch/headers.js:307]: Please keep this?

Comment (tsctx) [reply lib/fetch/headers.js:307]: Since there seemed to be no side effects, I did not consider it necessary.

**Comment by tsctx** [comment]
@KhafraDev 
Unit tests were added.

File: lib/fetch/headers.js (hunk 1 above)
Comment (tsctx) [reply lib/fetch/headers.js]: I think this change is not the purpose of this PR, so I will make the change in another PR.

**Pull request merged**